tests/unit/expire.tcl

```diff
@@ -833,3 +833,62 @@ start_server {tags {"expire"}} {
         assert_equal [r debug set-active-expire 1] {OK}
     } {} {needs:debug}
 }
+
+start_cluster 1 0 {tags {"expire external:skip cluster"}} {
+    test "expire scan should skip dictionaries with lot's of empty buckets" {
+        # Collect two slots to help determine the expiry scan logic is able
+        # to go past certain slots which aren't valid for scanning at the given point of time.
+        # And the next non empyt slot after that still gets scanned and expiration happens.
+
+        # hashslot(alice) is 749
+        r psetex alice 500 val
+
+        # hashslot(foo) is 12182
+        # fill data across different slots with expiration
+        for {set j 1} {$j <= 100} {incr j} {
+            r psetex "{foo}$j" 500 a
+        }
+        # hashslot(key) is 12539
+        r psetex key 500 val
+
+        assert_equal 102 [r dbsize]
+
+        # disable resizing
+        r config set rdb-key-save-delay 10000000
```

Comment:
How does this disable resizing?

Reply:
never ending bgsave forbids dict expansion/contraction.